app/models/chess_piece.rb

```diff
@@ -22,6 +24,20 @@ def horizontally_obstructed?(endpoint_x, endpoint_y)
     end
   end
 
+  def diagonally_obstructed?(endpoint_x, endpoint_y)
+    if endpoint_x > position_x
+      ((position_x + 1)..endpoint_x).find do |x|
+        position_y = x 
```

Comment:
@varghes1 what if `endpoint_x` is > `position_x` but `endpoint_y` is < `position_y`?